src/main/java/pl/cleankod/exchange/core/domain/Account.java

```diff
@@ -1,13 +1,16 @@
 package pl.cleankod.exchange.core.domain;
 
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import pl.cleankod.exchange.provider.AccountDeserializeService;
 import pl.cleankod.util.Preconditions;
 
 import java.util.UUID;
 import java.util.regex.Pattern;
 
+@JsonDeserialize(using = AccountDeserializeService.class)
```

Comment:
The domain layer should not be exposed to the presentation layer either directly or through the serializers